applications/main/archive/scenes/archive_scene_browser.c

```diff
@@ -34,6 +34,8 @@ const char* archive_get_flipper_app_name(ArchiveFileTypeEnum file_type) {
         return "Bad KB";
     case ArchiveFileTypeWAV:
         return EXT_PATH("apps/Media/wav_player.fap");
+    case ArchiveFileTypeMag:
+        return EXT_PATH("apps/GPIO/magspoof.fap");
```

Comment:
Also reminds me about the wav player, for the previous files we added support for, we moved the applications to the firmware repository. The idea was that if one day we were to make a build without apps included, these that have file associations would be included anyway. But now I'm not so sure, on both of these things. I'm not sure if we'd actually ever have a build without apps, sounds like unnecessary complexity and confusion, and would need some rework of the webupdater. And even if we do decide to do that, I'm not sure that including the apps regardless just because they have file associations, would make sense. Could just be that if you did install one without apps, then it would give an error for the app missing...

Nothing to change here, just a discussion we'll have to go over with the team

Reply:
Hm, I wonder if we could do something clever like implement an optional `ArchiveSupportedFaps` interface, like NFC does with `NfcSupportedCardsPlugin`, where one can populate the relevant stuct fields (just file icon and fap path?) and then the list gets auto-built onload of the archive. Probably out of scope, but would eliminate maintenance edits like this, and handle the above edge case you're mentioning nicely